core/deployment/src/main/java/io/quarkus/deployment/jvm/ResolvedJVMRequirements.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.deployment.jvm;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.jar.Attributes;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.builder.BuildException;
+import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.deployment.builditem.ModuleOpenBuildItem;
+
+/**
+ * Represents requirements and restrictions on the runtime.
+ * Currently only used to track add-opens requirements; we'd like to eventually
+ * support tracking, for example, the required JVM version as we start to see
+ * some libraries having more specific restrictions.
+ * Another use could be, for example, to force enabling experimental features such
+ * as needing jdk.incubator.vector and implied version requirements.
+ */
+public final class ResolvedJVMRequirements extends SimpleBuildItem {
```

Comment:
Why do we need an aggregated item for this? If new packaging types need to add support for these options, would they not have to add more methods to this? IMO the code to do this belongs with the packaging type.

Shared static utilities to construct the individual option sequences for JARs and JVMs would be more generally useful; locking these methods behind an instance creates a barrier that isn't really necessary IMO.